pkg/cache/defaulting_test.go

```diff
@@ -0,0 +1,290 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	fuzz "github.com/google/gofuzz"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestDefaultOpts(t *testing.T) {
+	t.Parallel()
+
+	pod := &corev1.Pod{}
+	testCases := []struct {
+		name string
+		in   Options
+
+		verification func(Options) string
+	}{
+		{
+			name: "ByObject.Namespaces gets defaulted from ByObject",
+			in: Options{
+				ByObject: map[client.Object]ByObject{pod: {
+					Namespaces: map[string]Config{
+						"default": {},
```

Comment:
Would it make sense to have cases where Config exists but its fields are nil?

Reply:
This is a testcase where config exists but its fields are nil :)

Reply:
... ups :) 

What I was thinking was basically did we test "both cases" based on your previous godoc comment:
> 	// Settings in the map value that are unset because either the value as a
>	// whole is nil or because the specific setting is nil will be defaulted.

But if I see correctly (now) "the value as a whole is nil" is not possible as our map values are all not pointers, right?

Reply:
Yeah, exactly. I did notice that discrepancy between godocs and code to and updated the godoc